service/javadsl/server/src/main/java/com/lightbend/lagom/javadsl/server/ServiceGuiceSupport.java

```diff
@@ -30,6 +33,138 @@
+     *
+     *
+     * <p>This method should be used when the {@link Router} has some dependencies and needs to get them injected
+     * by Lagom's runtime DI infrastructure (Guice).
+     *
+     * <p>Typically, this will be a {@link Router} generated from a Play {@code routes}
+     * or a akka-grpc generated Play {@link Router}.
+     *
+     * <p>Once you declare a {@link Router}, you may need to define it's prefix to indicate on which path it should be available.
+     *
+     * <pre>
+     * bindService(
+     *     MyService.class, MyServiceImpl.class,
+     *     additionalRouter(HelloWorldRouter.class).withPrefix("/hello"),
+     * );
+     * </pre>
+     *
+     * You don't need to configure a prefix if the {@link Router} has it pre-configured.
+     * A akka-grpc generated Play {@link Router}, for instance, has its prefix already defined by the gRPC descriptor
+     * and doesn't need to have its prefix reconfigured.
+     *
+     * <p>Note that this method won't create a binding and is intended to be used in conjunction with
+     * {@link ServiceGuiceSupport#bindService(Class, Service, AdditionalRouter, AdditionalRouter...)}
+     * or {@link ServiceGuiceSupport#bindService(Class, Class, AdditionalRouter, AdditionalRouter...)}.
+     * Calling this method outside this context will have no effect.
+     *
+     * @param router an additional Play Router class
+     * @return an AdditionalRouter instance
+     */
+    @ApiMayChange
+    default <R extends Router> AdditionalRouter additionalRouter(Class<R> router) {
+        return new ClassBased<R>(router);
+    }
+
+
+    /**
+     * Helper method to create an {@link AdditionalRouter} instance that can be used to
+     * declare additional Play {@link Router}s on a Lagom Service.
+     *
+     *
+     * <p>This method should be used when the {@link Router} does not have any
+     * other dependencies and therefore can be immediately passed as an instance.
+     *
+     * <p>Once you declare a {@link Router}, you may need to define it's prefix to indicate on which path it should be available.
+     *
+     * <pre>
+     * bindService(
+     *     MyService.class, MyServiceImpl.class,
+     *     additionalRouter(new HelloWorldRouter()).withPrefix("/hello"),
+     * );
+     * </pre>
+     *
+     * You don't need to configure a prefix if the {@link Router} has it pre-configured.
+     *
+     * <p>Note that this method won't create a binding and is intended to be used in conjunction with
+     * {@link ServiceGuiceSupport#bindService(Class, Service, AdditionalRouter, AdditionalRouter...)}
+     * or {@link ServiceGuiceSupport#bindService(Class, Class, AdditionalRouter, AdditionalRouter...)}.
+     * Calling this method outside this context will have no effect.
+     *
+     * @param router an additional Play Router instance
+     * @return an AdditionalRouter instance
+     */
+    @ApiMayChange
+    default AdditionalRouter additionalRouter(Router router) {
```

Comment:
Note, this special case accepts an `Router` instance and therefore I'm not mentioning the `routes` file nor `akka-grpc` here because it's not applicable. 

Both cases require DI are only relevant for the variant requiring a `.class`, not a `Router` instance.

Reply:
Do we have specific use cases that require the ability to pass a pre-constructed instance?

Reply:
Actually not. I started with an instance version to later realise that what I needed was one accepting a `Class` and supporting injection. 

It's always possible to write a router by hand that doesn't require any injection in which case it would be possible to just pass an instance. 

I would leave it for now since it's marked with `@ApiMayChange`. Chances are that we will replace it with your proposal of composite routers.

Reply:
> It's always possible to write a router by hand that doesn't require any injection in which case it would be possible to just pass an instance.

But again, even so, users can still pass a `Class`.